File: powersoft_ostec_customizations/doctype_changes/supplier_quotation_custom.py
import frappe
from frappe.utils import flt

def validate(self, method=None):
    copy_all_rates(self)

def on_submit(self, method=None):
	copy_all_rates(self)

def copy_all_rates(self):
    for row in self.items:
        row.custom_disti_quote = flt(row.rate)

File: powersoft_ostec_customizations/public/js/supplier_quotation_custom.js
frappe.ui.form.on('Supplier Quotation', {
	validate(frm) {
		copy_all_rates(frm)
	},
	on_submit(frm) {
		copy_all_rates(frm)
	},
})

frappe.ui.form.on('Supplier Quotation Item', {
	rate(frm, cdt, cdn) {
		copy_all_rates(frm)
	}
})


function copy_all_rates(frm){
    for (let row of frm.doc.items){
        row.custom_disti_quote = row.rate
    }
    frm.refresh_field("items")
}
